staker/stakerapp.go

```diff
@@ -506,6 +511,60 @@ func (app *App) mustSetTxSpentOnBtc(hash *chainhash.Hash) {
 	}
 }
 
+// SendPhase1Transaction receives the BTC staking transaction hash that
+// should be already in BTC and creates the necessary data to submit
+// the BTC delegation into the consumer chain
+func (app *App) SendPhase1Transaction(
+	stakerAddr btcutil.Address,
+	stkTxHash *chainhash.Hash,
+	tag []byte,
+	covenantPks []*secp256k1.PublicKey,
+	covenantQuorum uint32,
+) (btcDelegationTxHash string, err error) {
+	// check we are not shutting down
+	select {
+	case <-app.quit:
+		return "", nil
+	default:
+	}
```

Comment:
Not sure why this is being checked every time at the beginning and the end instead of only the end
any specific reason? @KonradStaniec

Reply:
this quick exit path, in case the application is already closing